backend/routes/userRoutes.js

```
@@ -1,93 +1,92 @@
-    },
-  });
+        username, 
+        email, 
+        password: hashedPasswordValue
+    }
+  })
 
   req.session.user = newUser;
 
-  res.json(newUser);
-});
+  res.json(newUser)
+})
 
-router.put("/:id", async (req, res) => {
-  const { id } = req.params;
-  const { username, email, password } = req.body;
+router.put('/:id', async(req, res) => {
+  const { id } = req.params
+  const {  username, email, password  } = req.body
 
   const updatedUser = await prisma.user.update({
     where: { id: parseInt(id) },
     data: {
-      username,
-      email,
-      password,
-    },
-  });
+        username, 
+        email, 
+        password
+    }
+  })
 
-  res.json(updatedUser);
-});
+  res.json(updatedUser)
+})
 
-router.delete("/:id", async (req, res) => {
-  const { id } = req.params;
+router.delete('/:id', async(req, res) => {
+  const { id } = req.params
 
   const deletedUser = await prisma.user.delete({
-    where: { id: parseInt(id) },
-  });
-
-  res.json(deletedUser);
-});
-
-router.post("/login", async (req, res) => {
-  const { username, password } = req.body;
-
-  try {
-    // Find the user by username
-    const user = await prisma.user.findFirst({ where: { username: username } });
-
-    if (!user) {
-      return res.status(401).json({ error: "Invalid username or password" });
+    where: { id: parseInt(id) }
+  })
+
+  res.json(deletedUser)
+
+})
+
+router.post('/login', async (req, res) => {
+    const { username, password } = req.body;
+
+    try {
+      // Find the user by username
+      const user = await prisma.user.findFirst({ where: { username: username } });
+  
+      if (!user) {
+        return res.status(401).json({ error: 'Invalid username or password' });
+      } 
+  
+      // Compare the password
+      const isValidPassword = await bcrypt.compare(password, user.password);
+  
+      if (!isValidPassword) {
+        return res.status(401).json({ error: 'Invalid username or password' });
```

Here we can throw "invalid password".

---

To do:
Re-work error messages